src/convenience/webhook.ts

```diff
@@ -91,7 +136,12 @@ export function webhookCallback<C extends Context = Context>(
             await bot.init();
             initialized = true;
         }
-        const { update, respond, end, handlerReturn } = server(...args);
+        const { update, respond, unauthorized, end, handlerReturn, header } =
+            server(...args);
+        if (header !== token) {
```

Comment:
shouldn't we rather use `token && header !== token`? to have a way to completely disable this check

Reply:
If Telegram delivers a token, but it is not supplied here, then there is a mismatch and the check will fail. Currently, the two configurations must match exactly.

What you are suggesting is to ignore if Telegram sends a token, is that better?